app/models/responses.go

```diff
@@ -80,20 +81,22 @@ func (r *ResultMessage) MarshalJSON() ([]byte, error) {
 	}
 
 	for _, attr := range r.MessageAttributes {
+		// When encoding json, BinaryValue([]byte) are automatically converted to base64.
+		binary, _ := base64.StdEncoding.DecodeString(attr.Value.BinaryValue)
 		m.MessageAttributes[attr.Name] = sqstypes.MessageAttributeValue{
 			DataType:    &attr.Value.DataType,
-			StringValue: &attr.Value.StringValue,
-			BinaryValue: []byte(attr.Value.BinaryValue),
+			StringValue: attr.Value.StringValue,
+			BinaryValue: binary,
```

Comment:
Here is the key change of this PR.

`encoding/json` lib automatically applies base 64 encode if the arg type is `slice`.
But our internal `MessageAttributeValue.BinaryValue` is already base 64 encoded string.
So the base64 encode was done "twice".

Reply:
I need some more information about what's going on here.  I don't really mind not encoding this, but I've gotten several complaints about the MessageAttributes in the last week or so and I need to be more careful about it.  

Can you explain more about what the issue is that you found, and how this will solve it?  Also the below if you know:

- When did the `[]byte` get encoded to begin with?  If we're doing it internally, let's just avoid doing it at all, and let the marshallers handle it.

- It's been a minute since I looked at this, but why did we have to do the override of `MarshalJSON` period here?  I see your comment about the structure, but I can't see why we needed an array to begin with.  It seems to me like it comes in as a map in `SendMessage` why not maintain the map format and not kill ourselves going back and forth?  Was this to be compatible with legacy code some how?  I know there's all this "convert to old style" stuff, but I sense that's the essence of a lot of these problems.

Reply:
The error we faced was like below: 
```
com.amazonaws.AmazonClientException: MD5 returned by SQS does not match the calculation on the original request. (MD5 calculated by the message attributes: "565a1514fba6a202b2f3e2de433300cf", MD5 checksum returned: "1dffa6b307b25550e81d62cf3377dacd")
```

> When did the []byte get encoded to begin with?
The base 64 encoding was performed by AWS SDK Client, and goaws preserves the base 64 encoded string received over HTTP for internal processing. So the value was already encoded at here.

> It's been a minute since I looked at this, but why did we have to do the override of MarshalJSON period here?
It's unclear whether we can completely get rid of MarshalJSON while supporting both xml and json, but as you say, tackling the below "old style" TODO will help simplify things.
https://github.com/Admiral-Piett/goaws/blob/master/app/utils/utils.go#L95

Reply:
Ok, I'm digging into this now a little more thoroughly as a result of this issue - https://github.com/Admiral-Piett/goaws/issues/328.  I think I understand what is going on in here now a little more.  

I think the overriding of the `MarshalJSON` method this way ties our hands, and there's a lot of different `"ResultMessageAttribute" VS "ResultAttribute"` stuff that makes this hard to consume.  I'm going to work on tweaking this a little and seeing if I can bring this piece more in line with the other end points.  Stand by!

Reply:
Ok, ok - I take it back.  🙃 .  The message attributes are a real mess and our flexibility is all tied up in this pattern, and to fix my issue I'll have to rebuild it all.  

If you, want I'll merge this unblock you if you're ready with all this, but I'm going to refactor the Message Attributes stuff now.  That will be my next order of business.  So after I do that, I'd appreciate it if you could smoke test it to make sure it still works for your team?

Thoughts on that plan?

Reply:
OK! We'd be happy to apply this change to resolve our block. 
And of course, let me help you on the test with your refactoring on MessageAttributes! 💪  We can use our app and AWS SDK for Java.

Reply:
As far as I look into #328, this PR won't resolve the issue with only itself. But it is related with jackson-ize process with empty map in go 🤔